archinstall/lib/installer.py

```diff
@@ -325,7 +325,7 @@ def post_install_check(self, *args :str, **kwargs :str) -> List[str]:
 
 	def enable_multilib_repository(self):
 		# Set up a regular expression pattern of a commented line containing 'multilib' within []
-		pattern = re.compile("^#\\[.*multilib.*\\]$")
+		pattern = re.compile("^#\\[.*multilib*\\]$")
```

Comment:
This regex will match:

* Beginning of line
* `#` character
* 0 or more whitespace characters
* `[` character
* 0 or more whitespace characters
* `multilib`
* 0 or more whitespace characters
* `]` character
* End of line

Using [raw string notation](https://docs.python.org/3/library/re.html) `r""` will also reduce the confusion of the regex backslashes.

Reply:
There is a **`*`** at the start of **`m`**, it will match till another **`*`** right? And there is no need to check whitespaces I think because if any whitespace is there, it will break `pacman.conf`

Reply:
```
"^#\\[.*multilib.*\\]$"
```

* `^` is start of line
* `#` is a character literal
* `\\[` is escaping the `[`, and `\\` is for escaping a `\` in a python string
* `.*` is a greedy match on all characters
* `multilib` is matching the literal characters
* `.*` is a greedy match on all characters
* `\\]` is escaping the `]`, and `\\` is for escaping a `\` in a python string
* `$` is end of line

-----

Changing to `"^#\\[.*multilib*\\]$"` will functionally change `multilib.*` to `multili[b]*`. It works because you are removing the second `.*`. It will no longer match `# [multilib-testing]`, but will match `# [multilib]` and `# [multili]`.

---

I wasn't able to find [an answer for whitespace in pacman.conf](https://archlinux.org/pacman/pacman.conf.5.html), but it seems like a safe assumption to make.

It could be simplified to `pattern = re.compile(r"^#\s*\[multilib\]$")`

Reply:
This is the simplest one, will test it out
Thanks for the details 🤗